src/nxt_http_rewrite.c

```diff
@@ -72,37 +71,16 @@ nxt_http_rewrite(nxt_task_t *task, nxt_http_request_t *r)
         return NXT_ERROR;
     }
 
-    p = (rp.args.length > 0) ? rp.args.start - 1 : rp.target_end;
-
-    encoded_path.start = rp.target_start;
-    encoded_path.length = p - encoded_path.start;
-
-    if (r->args->length == 0) {
-        r->target = encoded_path;
-
-    } else {
-        target.length = encoded_path.length + 1 + r->args->length;
-
-        target.start = nxt_mp_alloc(r->mem_pool, target.length);
-        if (target.start == NULL) {
-            return NXT_ERROR;
-        }
-
-        p = nxt_cpymem(target.start, encoded_path.start, encoded_path.length);
-        *p++ = '?';
-        nxt_memcpy(p, r->args->start, r->args->length);
-
-        r->target = target;
-        r->args->start = p;
-    }
-
     r->path = nxt_mp_alloc(r->mem_pool, sizeof(nxt_str_t));
     if (nxt_slow_path(r->path == NULL)) {
         return NXT_ERROR;
     }
 
     *r->path = rp.path;
 
+    r->quoted_target = rp.quoted_target;
+    r->uri_changed = 1;
+
     if (nxt_slow_path(r->log_route)) {
         nxt_log(task, NXT_LOG_NOTICE, "URI rewritten to \"%V\"", &r->target);
```

Comment:
Note to fix this.

Reply:
friends, just asking , any chance to get in in next release?

Reply:
This should definitely make 1.33, I'm just waiting for this commit to be appropriately split up before reviewing it.

Reply:
Yep, it will be in the next release.
We'll do it like this:
1. Andrei adds more tests for the proxy that has been in the https://github.com/nginx/unit/pull/1211.
2. I'll create a PR split from #1162.
3. Continue on the #1162 and related tests.